File: logger.pl
#!/usr/bin/env perl

use strict;
use warnings;
use Log::Log4perl;

my $log_layout = 'Log::Log4perl::Layout::PatternLayout';
my $log_layout_template = '%d{ISO8601} [%p] %m{chomp}%n';
my $log_file     = 'my_log.log';

my $log_conf = qq(
    log4perl.logger = DEBUG, Screen, File

    log4perl.appender.Screen = Log::Log4perl::Appender::Screen
    log4perl.appender.Screen.stderr = 0
    log4perl.appender.Screen.layout = $log_layout
    log4perl.appender.Screen.layout.ConversionPattern = $log_template
    log4perl.appender.Screen.Threshold = INFO

    log4perl.appender.File = Log::Log4perl::Appender::File
    log4perl.appender.File.mode = write
    log4perl.appender.File.filename = $log_file
    log4perl.appender.File.layout = $log_layout
    log4perl.appender.File.layout.ConversionPattern = $log_template
);

Log::Log4perl->init( \$log_conf );
my $logger = Log::Log4perl->get_logger();

File: logger.py
#!/usr/bin/env python

import logging

logfile = 'program.log'
logger = logging.getLogger()

# Root logger
logger.setLevel(logging.NOTSET)

# Console log handler
console = logging.StreamHandler()
console.setLevel(logging.INFO)
formatter = logging.Formatter(
    '%(asctime)s %(levelname)s %(message)s',
    "%Y-%m-%d-%H:%M:%s"    
)
console.setFormatter(formatter)
logger.addHandler(console)

# File log handler
file = logging.FileHandler(logfile)
file.setLevel(logging.DEBUG)
formatter = logging.Formatter(
    '%(asctime)s %(levelname)s %(message)s',
    "%Y-%m-%d-%H:%M:%s"    
)
file.setFormatter(formatter)
logger.addHandler(file)

logging.info("Info message goes to file and screen")
logging.debug("Debug message goes only to the file")
